spec/features/ajax_filters_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'Load filter options', type: :feature, js: true do
+  include_context :login_as_admin
+  before { visit cdrs_path }
+  context 'customer filter' do
+    let!(:goal_customer_list) {
+      [
+        FactoryBot.create(:customer, name: 'customer_1'),
+        FactoryBot.create(:customer, name: 'customer_2')
+      ]
+    }
+    let!(:other_customer) { FactoryBot.create(:customer, name: 'other') }
+
+    subject(:load_options) do |string = nil|
+      # open chosen popup to get input
+      find('#q_customer_id_chosen').click
+      find('#q_customer_id_chosen input').fill_in with: string || 'cus'
+    end
+
+    it 'should have one option in select' do
+      within '#sidebar #new_q #q_customer_input select', visible: :hidden do
+        expect(page).to have_css('option', count: 1, visible: :hidden)
+        expect(page).to have_css('option:first-child', text: 'Any', visible: :hidden)
+      end
+    end
+
+    it 'when type letters load options (2 results and "Any" default)' do
+      load_options
+      goal_customer_list.each do |item|
+        expect(page).to have_css('#sidebar #new_q #q_customer_input select option', text: item.name, visible: :hidden)
+      end
+      # 2 results from database and "Any" option by default = total 3
+      expect(page).to have_css('#sidebar #new_q #q_customer_input select option', count: 3, visible: :hidden)
+    end
+
+    it 'should select option and execute filter ' do
+      customer_name = goal_customer_list.first.name
+      load_options
+      # make first element selected
+      find('#q_customer_id_chosen li', text: customer_name).click
+
+      # execute filter and reload page
+      page.find('input[type=submit]').click
+
+      expect(page.current_url).to include('customer_id_eq')
+      # check selected element
+      result = find('#sidebar #new_q #q_customer_input select option', visible: :hidden, text: customer_name).selected?
+      expect(result).to eq true
```

Comment:
expect(find('#sidebar #new_q #q_customer_input select option', visible: :hidden, text: customer_name)).to be_selected